moped-editor/src/views/projects/projectView/ProjectComponents/DraftComponentListItem.js

```diff
@@ -6,15 +6,17 @@ import ListItemText from "@mui/material/ListItemText";
 import Cancel from "@mui/icons-material/Cancel";
 import CheckCircle from "@mui/icons-material/CheckCircle";
 import { COLORS } from "./mapStyleSettings";
+import { useComponentListItemText } from "./utils/componentList";
 
 const DraftComponentListItem = ({
-  primaryText,
-  secondaryText,
+  component,
   onSave,
   onCancel,
   saveButtonText,
   saveButtonDisabled,
 }) => {
+  const { primary, secondary } = useComponentListItemText(component);
```

Comment:
yay for having a hook ready to handle this.

i just had to code this out in a SQL view for the AGOL integration, and now it occurs to me that we should probably be using a generated column to create the `<name> - <subtype>` string at the DB level. not a big deal and i'm glad to see this hook reused 🚀